phpstan.neon

```diff
@@ -0,0 +1,15 @@
+includes:
+    - vendor/phpstan/phpstan-phpunit/extension.neon
+    - phpstan-baseline.neon
+
+parameters:
+    level: 3
+    paths:
+        - src
+        - tests
+    excludePaths:
+        - src/Transport/HttpAdapter.php
+        - src/Query/Match.php
```

Comment:
Are those too complex or hard to fix to pass the level=3 of PhpStan?

Reply:
`Match` is because of the name of the class is a reserved keyword with PHP 8+, so there is a syntax error for PHPStan.
And [`src/Transport/HttpAdapter.php`](https://github.com/ruflin/Elastica/blob/72a4598544e3f99b5dd8cacb05d009ee75c2a701/src/Transport/HttpAdapter.php) uses `egeloen/http-adapter` which is not installed so there are quite errors about unknown classes.

(both of them are deprecated)